app/src/main/java/com/nervousfish/nervousfish/service_locator/EntryActivity.java

```diff
@@ -42,19 +35,12 @@ protected void onResume() {
         super.onResume();
 
         final IServiceLocator serviceLocator = NervousFish.getServiceLocator();
-        final List<Profile> profiles = new ArrayList<>();
-        try {
-            final IDatabase database = serviceLocator.getDatabase();
-            profiles.addAll(database.getProfiles());
-        } catch (IOException e) {
-            LOGGER.error("IOException while getting profiles", e);
-        }
+        final Intent intent;
 
-        final Intent intent = new Intent();
-        if (profiles.isEmpty()) {
-            intent.setComponent(new ComponentName(this, WelcomeActivity.class));
+        if (serviceLocator.getDatabase().checkFirstUse()) {
+            intent = new Intent(this, LoginActivity.class);
         } else {
-            intent.setComponent(new ComponentName(this, LoginActivity.class));
+            intent = new Intent(this, WelcomeActivity.class);
```

Comment:
Can you restore the code as it was before (with the `final Intent intent = new Intent();`)